ftw/linkchecker/report_mailer.py

```diff
@@ -9,41 +9,45 @@
 import plone.api
 
 
-app = globals()['app']
-app = makerequest(app)
-
-PLONE = app.get('Plone')
-setSite(PLONE)
-
-
-def send_feedback(email_subject, email_message, receiver_email_address, report_path):
-    """Send an email including an excel workbook attached.
-    """
-    mh = plone.api.portal.get_tool('MailHost')
-    from_name = PLONE.getProperty('email_from_name', '')
-    from_email = PLONE.getProperty('email_from_address', '')
-
-    sender = 'Linkcheck Reporter'
-    recipient = from_email
-    file_name = report_path.rsplit('/', 1)[1]
-
-    msg = MIMEMultipart()
-    msg['From'] = "%s <%s>" % (from_name, from_email)
-    msg['reply-to'] = "%s <%s>" % (sender, recipient)
-    msg['To'] = receiver_email_address
-    msg['Date'] = formatdate(localtime=True)
-    msg['Subject'] = Header(email_subject, 'windows-1252')
-    msg.attach(MIMEText(email_message.encode(
-        'windows-1252'), 'plain', 'windows-1252'))
-    part = MIMEBase('application', "octet-stream")
-    part.set_payload(open(report_path, "rb").read())
-    encoders.encode_base64(part)
-    part.add_header(
-        'Content-Disposition',
-        'attachment; filename="' + file_name + '"'
-    )
-    msg.attach(part)
-
-    mh.send(msg, mto=receiver_email_address,
-            mfrom=from_email,
-            immediate=True)
+class MailSender(object):
+
+    def __init__(self, portal=None):
+        if not portal:
+            self.app = globals()['app']
+            self.app = makerequest(self.app)
+            self.PLONE = self.app.get('Plone')
+        if portal:
+            self.PLONE = portal
+        setSite(self.PLONE)
+
+    def send_feedback(self, email_subject, email_message, receiver_email_address, report_path):
+        """Send an email including an excel workbook attached.
+        """
+        mh = plone.api.portal.get_tool('MailHost')
+        from_name = self.PLONE.getProperty('email_from_name', '')
+        from_email = self.PLONE.getProperty('email_from_address', '')
+
+        sender = 'Linkcheck Reporter'
+        recipient = from_email
+        file_name = report_path.rsplit('/', 1)[1]
```

Comment:
There is rstrip for this purpose --> https://www.tutorialspoint.com/python/string_rstrip.htm

Reply:
rstrip would remove a string from the right end. What I want to do is get the filename from path like:
```
>>> string = '/some/path/filename.xmlx'
>>> string.rsplit('/', 1)[1]
'filename.xmlx'
```

Reply:
I close this conversation since this is going to be removed as soon as I'm using tmpfile